src/app/rundown/components/part/part.component.ts

```diff
@@ -1,42 +1,77 @@
-import { Component, EventEmitter, Input, Output } from '@angular/core'
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core'
 import { Part } from '../../../core/models/part'
 import {animate, state, style, transition, trigger} from '@angular/animations';
+import { Piece } from '../../../core/models/piece'
+import { PieceLayerService } from '../../../shared/services/piece-layer.service'
+import { PieceLayer } from '../../../shared/enums/piece-layer'
 
 @Component({
   selector: 'sofie-part',
   templateUrl: './part.component.html',
-  styleUrls: ['./part.component.scss'],
-  animations: [
-    trigger('onAirTrigger', [
-      state('onAir', style({
-        backgroundColor: 'red'
-      })),
-      state('offAir', style({
-
-      })),
-      state('isNext', style({
-        backgroundColor: 'green'
-      })),
-      transition('offAir => onAir', animate(500)),
-      transition('offAir => isNext', animate(500)),
-      transition('onAir => isNext', animate(500))
-    ]),
-  ]
+  styleUrls: ['./part.component.scss']
 })
-export class PartComponent {
+export class PartComponent implements OnChanges {
+  // TODO: Figure out how to "subscribe" to changes for this on the part property.
```

Comment:
That is a solution, but I think it will be hard to track what happens where in the application.
I think the proper way to handle this is to do immutable "updates" to the entity state variables.

Reply:
That also sounds like a viable solution :)